pkgs/development/python-modules/pyecotrend-ista/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, setuptools
+, dataclasses-json
+, requests
+, pytestCheckHook
+, pytest-cov
+, pytest-dataset
+, pytest-xdist
+, requests-mock
+, syrupy
+}:
+
+buildPythonPackage rec {
+  pname = "pyecotrend-ista";
+  version = "3.3.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.11";
+
+  src = fetchFromGitHub {
+    owner = "Ludy87";
+    repo = "pyecotrend-ista";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-NAJ3/37dlwOiP1cilxfADUcuUt1VKF+eo/3JKgSUo3M=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    dataclasses-json
+    requests
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-cov
```

Comment:
We don't care about test coverage. Remove the dependency and patch it out

```nix
  src = {
    ...
  };

  postPatch = ''
    sed -i "/addopts =/d" pyproject.toml
  '';
```

https://github.com/Ludy87/pyecotrend-ista/blob/main/pyproject.toml#L24